### PR TITLE
feat: Private attachments

### PR DESCRIPTION
**Pull request by tommoor**
Continuing the work in https://github.com/outline/outline/pull/1137

### CODE REVIEW

**Comment by tommoor** [comment]
@HUSSTECH I'm not able to PR into yours because it's in a different repo, but I have added support for public attachments for avatar uploads here.

Just public shared documents to do.

**Comment by tommoor** [comment]
...aaaand now with support for publicly shared documents. I think this covers the final pieces.

File: server/presenters/document.js [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
 import presentUser from './user';
 
 type Options = {
   isPublic?: boolean,
 };
 
+const attachmentRegex = /!\[.*\]\(\/api\/attachments\.redirect\?id=(?<id>.*)\)/gi;
+
+// replaces attachments.redirect urls with signed/authenticated url equivalents
+async function replaceImageAttachments(text) {
```
Comment (tommoor) [inline server/presenters/document.js]: Not sure if it's worth putting this utility method elsewhere? It's only used here so seems to make sense to co-locate.

Comment (HUSSTECH) [reply server/presenters/document.js]: it is pretty much encapsulated perfectly in here. No other "unit" is publicly shareable except for Documents, so I think it's good here. Also definitely much neater to have it in the `presenter` module.

**Pull request closed without merging**